ttnn/cpp/ttnn/operations/reduction/topk/device/kernels/dataflow/reader_create_index_local_topk.cpp

```diff
@@ -0,0 +1,75 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+#include "dataflow_api.h"
+/**
+ * add a cb full of indices for the tile
+ * each row is identical in the index tensor, so we just need to add an offset based on which row tile it is
+ * first 32 elements are {0,..31}, then next 32 are {32,..64}
+ * wt is which tile it is along the row [0, Wt) so j + 32*wt is the value in the tile at each element
+*/
+FORCE_INLINE void generate_index_tile(const uint32_t cb_id, const uint32_t wt) {
+    // TODO: investigate moving to compile time (binary size is at risk)
+    cb_reserve_back(cb_id, 1);
+    uint32_t write_addr = get_write_ptr(cb_id);
+    volatile tt_l1_ptr uint32_t* ptr = reinterpret_cast<volatile tt_l1_ptr uint32_t*>(write_addr);
+    uint16_t wt_offset = wt << 5;
+
+    uint32_t count = 0;
+    for (uint32_t i = 0; i < 2; ++i) {
+        for (uint32_t j = 0; j < 2; ++j) {
+            for (uint32_t k = 0; k < 16; ++k) {
+                for (uint32_t l = 0; l < 16; l+=2) {
+                    uint16_t value = l + 16*j + wt_offset;
+                    ptr[count] = (value + 1) << 16 | value;
+                    count++;
+                }
+            }
+        }
+    }
+    cb_push_back(cb_id, 1);
+}
+
+
+void kernel_main() {
+    uint32_t src_addr  = get_arg_val<uint32_t>(0);
+    uint32_t start_ht = get_arg_val<uint32_t>(1);
+    uint32_t start_wt = get_arg_val<uint32_t>(2);
+
+
+    constexpr uint32_t cb_id_in0 = get_compile_time_arg_val(0);
+    constexpr uint32_t cb_id_in1 = get_compile_time_arg_val(1);
+    constexpr bool src_is_dram = get_compile_time_arg_val(2) == 1;
```

Comment:
`... = (bool)get_compile_time_arg_val(2)` is cleaner

Reply:
Done